src/MICore/Transports/LocalUnixTerminalTransport.cs

```diff
@@ -121,6 +124,8 @@ public override void Close()
             // If we are shutting down before the _exitReader has read a line it's possible that
             // there is a thread blocked doing a read() syscall. 
             ForceDisposeStreamReader(_exitReader);
+            _streamReadPidCancellationTokenSource.Cancel();
+            _streamReadPidCancellationTokenSource.Dispose();
```

Comment:
If you hit stop debugging while this is up, are we far enough along to get to Close?